### PR TITLE
Improve Custodia key distribution handling, by ensuring that upon replica installation the keys are  successfully shared with the peer systems.

### PR DESCRIPTION
**Pull request by ndehadrai**
Test for improved Custodia key distribution

The test checks that custodia keys are properly replicated from the source and are successfully distributed amongst peer system upon successful replica installation.

Fixes:  https://pagure.io/freeipa/issue/7518

Signed-off-by: Nikhil Dehadrai <ndehadra@redhat.com>

### CODE REVIEW

**Comment by tiran** [comment]
Hi @ndehadrai,

thanks for your contribution. Two files are conflicting. Please rebase your patch on master.

**Comment by stlaz** [comment]
(the commit(s) should also have a proper message and description)

File: ipatests/test_integration/test_replica_promotion.py [hunk 1]
```diff
@@ -660,3 +660,37 @@ def test_sign_with_subca_on_replica(self):
...
+                              setup_dns=True)
+        replica1.run_command(['ipactl', 'status'])
+        replica1.run_command(['systemctl', 'stop', 'ipa-custodia'])
+        replica1.run_command(['ipactl', 'status'])
+
+        # Install Replica2 with CA with source as Replica1.
+        result1 = tasks.install_replica(replica1, replica2, setup_ca=True,
+                                        setup_dns=True)
+        replica2.run_command(['ipactl', 'status'])
+        assert(result1.returncode == 0), (
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: The ``assert`` is not necessary. ``run_command`` raises an exception when the command fails with a non-zero return code.

File: ipatests/test_integration/test_replica_promotion.py [hunk 2]
```diff
@@ -660,3 +660,37 @@ def test_sign_with_subca_on_replica(self):
...
+        # Install Replica1 without CA and stop ipa-custodia
+        tasks.install_replica(master, replica1, setup_ca=False,
+                              setup_dns=True)
+        replica1.run_command(['ipactl', 'status'])
+        replica1.run_command(['systemctl', 'stop', 'ipa-custodia'])
+        replica1.run_command(['ipactl', 'status'])
+
+        # Install Replica2 with CA with source as Replica1.
+        result1 = tasks.install_replica(replica1, replica2, setup_ca=True,
+                                        setup_dns=True)
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: Why do you install with DNS?

File: ipatests/test_integration/test_replica_promotion.py [hunk 3]
```diff
@@ -660,3 +660,37 @@ def test_sign_with_subca_on_replica(self):
...
+        tasks.install_master(cls.master, domain_level=cls.domain_level)
+
+    def test_replica_install_for_custodia(self):
+        master = self.master
+        replica1 = self.replicas[0]
+        replica2 = self.replicas[1]
+
+        # Install Replica1 without CA and stop ipa-custodia
+        tasks.install_replica(master, replica1, setup_ca=False,
+                              setup_dns=True)
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: Why DNS?

**Comment by tiran** [comment]
Please change the commit message, too. The first line should be a short title, then an empty line followed by an explanation.

**Comment by ndehadrai** [comment]
@tiran Can you please review the changes

**Comment by tiran** [comment]
@ndehadrai The commit message of commit https://github.com/freeipa/freeipa/pull/2027/commits/41f2606e17761263f9586eed5834bb303d86f598 doesn't follow FreeIPA's style for commit messages. Please follow my suggestion from comment https://github.com/freeipa/freeipa/pull/2027#pullrequestreview-130792645

**Comment by rcritten** [comment]
master:

* dcaa62f6a4e0a57de9d0affda584a27539bc4a36 Test for improved Custodia key distribution

**Pull request closed without merging**